src/Oscoin/Protocol/Sync/RealWorld.hs

```diff
@@ -227,33 +229,20 @@ runSync syncContext (SyncT s) = do
       Right r  -> pure r
 
 -- | Syncs a node, in a monad which can do @IO@.
+-- This function is implemented as an endless loop. During each loop, we check
+-- whether or not we are done syncing (cfr. 'isDone') and, if we are not, we
+-- sync the required blocks, wait 30 seconds, and try again.
 syncNode
     :: forall c tx s m.
        ( MonadIO m
        , Ord tx
        , Ord s
        , Ord (Hash c)
        , Ord (Beneficiary c)
+       , Buildable (Hash c)
        )
     => SyncT c tx s m ()
-syncNode = do
-    ctx <- ask
-    retrying policy (checkSynced ctx) $ \ _retryStatus -> do
-        remoteTip <- getRemoteTip
-        localTip  <- lift (BlockStore.getTip (scLocalChainReader ctx))
-        for_ (range localTip remoteTip) syncBlocks
-  where
-      policy :: RetryPolicyM (SyncT c tx s m)
-      policy = exponentialBackoff 100_000
-
-      -- Retry syncing if 'isDone' returns False.
-      checkSynced
-          :: SyncContext c tx s m
-          -> RetryStatus
-          -> ()
-          -> SyncT c tx s m Bool
-      checkSynced ctx _retryStatus () = do
-          -- We need to refetch the tips.
-          remoteTip <- getRemoteTip
-          localTip  <- lift (BlockStore.getTip (scLocalChainReader ctx))
-          pure $ not (isDone (scNu ctx) localTip remoteTip)
+syncNode = forever $ do
+    sync `catchError` \(e :: SyncError) ->
+       recordEvent $ Telemetry.Events.NodeSyncError (toException e)
+    liftIO $ threadDelay 30000000 -- Wait 30 seconds.
```

Comment:
What's the idea here with waiting 30s?

Reply:
Mainly just as throttling mechanism -- I have picked a number out of thin air, we can either make this configurable or to pick something different.

Reply:
91af9d63